pkg/traceql/ast_execute.go

```diff
@@ -250,7 +250,7 @@ func (a Aggregate) evaluate(input []*Spanset) (output []*Spanset, err error) {
 			count := 0
 			for _, s := range ss.Spans {
 				val, err := a.e.execute(s)
-				if err != nil {
+				if err != nil || val.Equals(&StaticNil) {
```

Comment:
is this related to symmetric traceql? i'm not opposed. it makes sense to ignore nils here but i wonder if this is part of a larger change to have better nil handling across traceql.

Reply:
After making the nil check in `isMatchingOperand()` symmetric the tests for the query `{{} | avg(.dne) != 0}` failed, because the aggregation over the non existing attribute returned results. The explanation is that parts of the logic relied on the asymmetric nil check in `isMatchingOperand()`

> i wonder if this is part of a larger change to have better nil handling across traceql

Good point. I'll run some more tests and look out for other missing nil checks.

Reply:
I've checked the aggregates `min`, `max`, and `sum` and they now all return results when they are applied to non existing attributes like this `{} | sum(.dne) != 0}`.

![image](https://github.com/user-attachments/assets/1339b23b-4032-41b2-8c2b-a1c207066d27)

Since there is a test for this in `searchesThatDontMatch` [here](https://github.com/grafana/tempo/blob/d6b6e2d8383c9fdee4a549e7b6cd30d9d555570f/tempodb/tempodb_search_test.go#L276), I assume that the correct behavior is that all those aggregations don't return results.

On the other hand: looking at the results above the `sum(.dne)` column is populated with `nil` and technically `nil != 0` is true. Therefore returning results is technically also not wrong. I would just like to confirm the correct behavior here...

Reply:
The behavior that I've now implemented for `avg`, `sum`, `min`, and `max` in [this commit](https://github.com/grafana/tempo/pull/4869/commits/d2517d6e8954ba16f223aa1bc1264dfa6560ed4c) is as follows:
* Spans without the attributes are skipped from the aggregate
* Spansets without any any span with the aggregated attribute are removed from the results

Reply:
@electron0zero was asking questions about nil handling awhile back and we never answered that Q. I personally like the behavior you've chosen. I think it's the most consistent with the language.